packages/eslint-plugin-clarity-migration/src/rules/utils.ts

```diff
@@ -0,0 +1,56 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+
+function isProperty(node: TSESTree.Node): node is TSESTree.Property {
+  return node.type === 'Property';
+}
+
+function isIdentifier(node: TSESTree.Node): node is TSESTree.Identifier {
+  return node.type === 'Identifier';
+}
+
+function isCallExpression(node: TSESTree.Node): node is TSESTree.CallExpression {
+  return node.type === 'CallExpression';
+}
+
+function isObjectExpression(node: TSESTree.Node): node is TSESTree.ObjectExpression {
+  return node.type === 'ObjectExpression';
+}
+
+export const getDecoratorProperty = (decorator: TSESTree.Decorator, name: string): TSESTree.Property | undefined => {
+  const arg = getDecoratorArgument(decorator);
+
+  if (!arg || !isObjectExpression(arg)) {
+    return undefined;
+  }
+
+  const properties = arg.properties as Array<TSESTree.Property>;
+  const property = properties.find(prop => !!(prop.key && isIdentifier(prop.key) && prop.key.name === name));
+
+  if (!property || !isProperty(property)) {
+    return undefined;
+  }
+
+  return property;
+};
+
+export const getDecoratorPropertyValue = (
+  decorator: TSESTree.Decorator,
+  name: string
+): TSESTree.Expression | TSESTree.Literal | undefined => {
+  const property = getDecoratorProperty(decorator, name);
+  if (!property) {
+    return undefined;
+  }
+
+  return property.value as any;
```

Comment:
I am curious - Why we need to cast? I mean we already specify what type of values we expect in line 39.

Reply:
Basically - mismatching types from `@typescript-eslint` and `estree`.